frontend/apps/reader/modules/readerlink.lua

```diff
@@ -662,19 +678,41 @@ function ReaderLink:onTap(_, ges)
     end
 end
 
---- Remember current location so we can go back to it
-function ReaderLink:addCurrentLocationToStack()
+function ReaderLink:getCurrentLocation()
+    local location
     if self.ui.document.info.has_pages then
-        table.insert(self.location_stack, self.ui.paging:getBookLocation())
+        location = self.ui.paging:getBookLocation()
     else
-        table.insert(self.location_stack, {
-            xpointer = self.ui.rolling:getBookLocation(),
-        })
+        location = {xpointer = self.ui.rolling:getBookLocation(),}
     end
+    return location
+end
+
+-- Returns true, current_location if the current location is the same as the
+-- saved_location on the top of the stack.
+-- Otherwise returns false, current_location
+function ReaderLink:compareLocationToCurrent(saved_location)
+    local current_location = self:getCurrentLocation()
+    if self.ui.rolling and saved_location.xpointer
+    and saved_location.xpointer == current_location.xpointer then
```

Comment:
This (multiline `if` conditions with no indentation) reads suprisingly well, but it's a bit unusual, I don't think we have any that looks like that, so more indentation please, or all on a single line :)

Reply:
I think it's come up before (possibly due to the same author :-) ).

My preference for multiline goes toward:
```lua
if
  this
  and that
end
```

Reply:
Yeah, we already discussed our preferences many times :) I don't even remember which I stated was mine :)
But as we don't have one enforced by our style guide, and various ones may be fine depending on context/lenghts/logic, I  think it could still come done down to one rule:
In an existing file, try to use the same style (that the elders introduced :) that is used around in this file, just so at least reading this file is consistent.

Reply:
Yes, that's generally what I try to do unless it's for git blame reasons or something (e.g., making something multiline that wasn't).